Elasticsearch configuration strategy support (#507)
elastic/terraform-provider-ec: docs/resources/ec_deployment.md

* `trust_all` (Optional) If true, all clusters in this external entity will be trusted and the `trust_allowlist` is ignored.
* `trust_allowlist` (Optional) The list of clusters to trust. Only used when `trust_all` is `false`.

##### Strategy

> The optional `elasticsearch.strategy` allow to choose the configuration strategy used to apply the changes.

Reviewer: IMO we should have something in here along the lines of "don't change this unless you know that you need to", it's not functionality customers should be concerned with.

Author: I have added kind of a warning.

> The optional `elasticsearch.strategy` allows choosing the configuration strategy used to apply the changes. This is a fine tune setting, you do not change it unless you have a specific case where the `autodetect` do not cover your use case.

Reviewer 2: ```suggestion
The optional `elasticsearch.strategy` allows you to choose the configuration strategy used to apply the changes. You do not need to change this setting unless you have a specific case where the `autodetect` does not cover your use case.
```
